forge/src/main/java/org/valkyrienskies/mod/forge/compat/TurtleReenterCommand.java

```diff
@@ -0,0 +1,60 @@
+package org.valkyrienskies.mod.forge.compat;
+
+import dan200.computercraft.api.turtle.ITurtleAccess;
+import dan200.computercraft.api.turtle.ITurtleCommand;
+import dan200.computercraft.api.turtle.TurtleCommandResult;
+import dan200.computercraft.shared.turtle.core.InteractDirection;
+import java.util.List;
+import net.minecraft.core.BlockPos;
+import net.minecraft.world.level.Level;
+import org.jetbrains.annotations.NotNull;
+import org.joml.Vector3d;
+import org.valkyrienskies.core.api.Ship;
+import org.valkyrienskies.mod.common.VSGameUtilsKt;
+import org.valkyrienskies.mod.common.config.VSGameConfig;
+
+public class TurtleReenterCommand implements ITurtleCommand {
+    private InteractDirection direction;
+
+    public TurtleReenterCommand(InteractDirection inputDirection) {
+        direction = inputDirection;
+    }
+
+    @NotNull
+    @Override
+    public TurtleCommandResult execute(@NotNull ITurtleAccess iTurtleAccess) {
+        if (VSGameConfig.SERVER.getCOMPUTERCRAFT().getTurtlesCanReenterShip()) {
+            BlockPos currentPos = iTurtleAccess.getPosition();
+            BlockPos potentShipPos = currentPos.relative(direction.toWorldDir(iTurtleAccess));
+            Level world = iTurtleAccess.getWorld();
+
+            Vector3d shipPos = getShipPosFromWorldPos(world, potentShipPos);
+            Ship ship = getShip(world, shipPos);
+            if (ship != null) {
+                if (iTurtleAccess.teleportTo(world, new BlockPos(shipPos.x + 0.5, shipPos.y + 0.5, shipPos.z + 0.5))) {
+                    Object[] results = new Object[1];
+                    results[0] = "reentered";
+                    return TurtleCommandResult.success(results);
+                }
+                return TurtleCommandResult.failure("obstructed");
+            }
+            return TurtleCommandResult.failure("no ship");
+        }
+        return TurtleCommandResult.failure("disabled");
+    }
+
+    private static Vector3d getShipPosFromWorldPos(Level world, BlockPos position) {
```

Comment:
This doesn't seem correct. `transformToNearbyShipsAndWorld` will return a pos in world space if `position` is ship space. Also why does it ignore the list and only use the first "ship" / world pos returned?

Reply:
Basically,  i needed to grab whatever ship was in the direction of the turtle (front, up, down) to then "reenter" the ship (aka move the turtle from world space to the closest ship space position IF there was a ship in front of it

In case multiple ships are infront of the turtle, itll just grab the first one

Reply:
I think for that to work you should use `getShipDataIntersecting()` instead to find ships that intersect with the AABB of the turtle.

Then you have to choose which ship the turtle moves to somehow; maybe you could make the command take in a ship name, and make the turtle either move to the first ship found, or the ship with the same name.